### ISSUE TITLE
Bug in TimeSeriesKMeans using metric 'softdtw'

### ISSUE THREAD

**Issue by nikhilRP** [rtavenar/tslearn#38]
Hello @rtavenar ,

I was trying out the TimeSeriesKMeans model with variable length time series.
Unfortunately the model seems to fail using "softdtw" as a metric.

Code I tried out.

```
import numpy
import random

from tslearn.clustering import TimeSeriesKMeans

seed = 0
numpy.random.seed(seed)

ts = list()

# random time series of variable length
ts.append(random.sample(range(1, 1000), 100))
ts.append(random.sample(range(1, 1000), 99))
ts.append(random.sample(range(1, 1000), 98))

# Soft-DTW-k-means
print("Soft-DTW k-means")
sdtw_km = TimeSeriesKMeans(
    n_clusters=2, metric="softdtw", verbose=True, random_state=seed)
y_pred = sdtw_km.fit_predict(ts)
```

`ValueError: Input contains NaN, infinity or a value too large for dtype('float64').`

Any help would be appreciated, thanks.

**Comment by rtavenar** [comment]
Hi @nikhilRP , and thanks for this very precise bug report.

I'll try to investigate on this as soon as I can. This is definitely a bug, but I have no idea at the moment how difficult it would be to solve it.
I'll let you know.

**Comment by rtavenar** [comment]
@nikhilRP 

I have just pushed a version that fixes the issue on my side. Coud you confirm that it's OK for you? If so, I'll add a doctest to ensure that we do not have the same bug in later versions.

Best,
Romain

**Comment by nikhilRP** [comment]
Thanks for taking time and fixing it. I did test the fix and it works fine, very much appreciated.

**Comment by rtavenar** [comment]
Great! By the way, I guess this fix is a step forward regarding Issue #16 :)
